packages/peregrine/lib/talons/SignIn/useSignIn.js

```diff
@@ -33,28 +35,24 @@ export const useSignIn = props => {
         { getUserDetails, setToken }
     ] = useUserContext();
 
-    const [signIn, { error: signInError }] = useMutation(signInMutation, {
-        fetchPolicy: 'no-cache'
+    const [signIn] = useMutation(signInMutation, {
+        fetchPolicy: 'no-cache',
+        onError: signInError => {
+            handleFormErrors([signInError]);
```

Comment:
The new FormErrors component removes the need to do much of this manual handling. You would just return the errors in a form errors array and then pass that to your `<FormErrors>` component:

https://github.com/magento/pwa-studio/blob/4e65464b1f77e19939ff46ef0f91cdcb11fc3271/packages/peregrine/lib/talons/CheckoutPage/ShippingInformation/AddressForm/useCustomerForm.js#L135

Reply:
I am doing this instead of a simple `formErrors: [error1, error2]` because these states are being lifted on to the parent. The parent has no idea which one's a duplicate error, all it does it append it the list of errors it maintains. If I keep this login inside a `useEffect` and check against the error from `const [signIn, { error: 'signInError' }]`, I have maintain another variable to not push duplicate values. To avoid doing that, I have used the `onError` structure on `useMutation` which will only fire once on an error occurrence.

Reply:
What component is the parent and why does it need to know about errors from the sign in form?